src/invidious/jsonify/api_v1/video_json.cr

```diff
@@ -79,72 +79,67 @@ module Invidious::JSONify::APIv1
+              json.field "targetDurationSec", fmt.target_duration if fmt.target_duration
+              json.field "maxDvrDurationSec", fmt.max_dvr_duration if fmt.max_dvr_duration
 
               # Not available on MPEG-4 Timed Text (`text/mp4`) streams (livestreams only)
-              json.field "bitrate", fmt["bitrate"].as_i.to_s if fmt["bitrate"]?
+              json.field "bitrate", fmt.bitrate.to_s if fmt.responds_to?(:bitrate)
 
               if proxy
                 json.field "url", Invidious::HttpServer::Utils.proxy_video_url(
-                  fmt["url"].to_s, absolute: true
+                  fmt.url, absolute: true
                 )
               else
-                json.field "url", fmt["url"]
+                json.field "url", fmt.url
               end
 
-              json.field "itag", fmt["itag"].as_i.to_s
-              json.field "type", fmt["mimeType"]
-              json.field "clen", fmt["contentLength"]? || "-1"
+              json.field "itag", fmt.itag.to_s
+              json.field "type", fmt.raw_mime_type
+              json.field "clen", fmt.content_length if fmt.responds_to?(:content_length)
+
+              json.field "encoding", fmt.codecs
 
               # Last modified is a unix timestamp with µS, with the dot omitted.
               # E.g: 1638056732(.)141582
               #
               # On livestreams, it's not present, so always fall back to the
               # current unix timestamp (up to mS precision) for compatibility.
-              last_modified = fmt["lastModified"]?
-              last_modified ||= "#{Time.utc.to_unix_ms.to_s}000"
-              json.field "lmt", last_modified
+              last_modified = fmt.last_modified || Time.utc
+              json.field "lmt", "#{last_modified.to_unix_ms}000"
 
-              json.field "projectionType", fmt["projectionType"]
+              json.field "projectionType", fmt.projection_type.to_s.upcase
 
-              if fmt_info = Invidious::Videos::Formats.itag_to_metadata?(fmt["itag"])
-                fps = fmt_info["fps"]?.try &.to_i || fmt["fps"]?.try &.as_i || 30
-                json.field "fps", fps
+              if fmt.is_a?(Videos::AdaptativeVideoStream)
+                json.field "fps", fmt.video_fps
+                json.field "size", "#{fmt.video_width}x#{fmt.video_height}"
+              end
+
+              if fmt_info = Invidious::Videos::Formats.itag_to_metadata?(fmt.itag)
                 json.field "container", fmt_info["ext"]
-                json.field "encoding", fmt_info["vcodec"]? || fmt_info["acodec"]
 
                 if fmt_info["height"]?
                   json.field "resolution", "#{fmt_info["height"]}p"
 
                   quality_label = "#{fmt_info["height"]}p"
-                  if fps > 30
-                    quality_label += "60"
-                  end
+                  quality_label += "60" if fmt.responds_to?(:video_fps) && fmt.video_fps > 30
```

Comment:
Please use the fps value instead of hardcoding it to 60, as 50fps streams exist too.